.gitignore

```diff
@@ -1,4 +1,5 @@
 .idea/*
 vendor/*
 build/
-.DS_Store
\ No newline at end of file
+.DS_Store
+composer.lock
```

Comment:
Not for libraries: https://getcomposer.org/doc/02-libraries.md#lock-file
Committing lock file is relevant (and highly recommended) at the _project_ level, so that all developers + production get the same exact versions of dependencies. 
At the _library_ level, dependencies' lock files are ignored by composer when running `composer install` from the project (only the project's `composer.lock` will be read). 
So in our case, committing it has no effect, and may even mislead your co-maintainers thinking the library works with  `foo/dependency:^1.1` because `1.1.0` is locked, though everyone using it (at project level) use `foo/dependency:^1.8` for instance (and maybe there are significant changes between 1.1 and 1.8, that your CI won't be aware of if you lock the dependency version).